zpy/objects.py

```diff
@@ -586,3 +586,44 @@ def restore_pose(
     obj = verify(obj)
     log.info(f"Restoring pose {pose_name} to object {obj.name}")
     obj.matrix_world = _SAVED_POSES[pose_name]
+
+
+def lighting_randomize(
+    path: Union[Path, str] = "hdri",
+    energy_jitter: bool = True,
+) -> None:
+    """Randomizes Lighting Types.
+
+    Args:
+        path: str: Where to pull HDRI textures from
+        energy_jitter: bool: Whether to jitter the lighting intensity for bounded and sun lights. Defaults to True
+    """
+
+    bounded_lights = []
+    sun_lights = []
+    switch = [True, False]
+
+    for obj in bpy.data.lights:
+        if obj.type == "POINT" or obj.type == "SPOT" or obj.type == "AREA":
+            log.info("BOUNDED light found")
+            bounded_lights.append(obj.name)
+        if obj.type == "SUN":
+            log.info("SUN light found")
+            sun_lights.append(obj.name)
+        elif bpy.data.lights == None:
+            log.debug("add lights to use this function")
+
+        # randomly hide lights
+        for obj in sun_lights:
```

Comment:
This should be outside the `for obj in bpy.data.lights` loop. Alternatively, rather than (1) looping through all the lights and putting them into lists and then looping through those lists, it might be faster to (2) randomize the `hide_render` and `data.energy` when looping through the `bpy.data.lights` directly.